pkg/controller/elasticsearch/elasticsearch_controller.go

```diff
@@ -216,6 +216,12 @@ func (r *ReconcileElasticsearch) Reconcile(request reconcile.Request) (reconcile
 
 	state := esreconcile.NewState(es)
 	results := r.internalReconcile(es, state)
+	if results.HasError() {
+		res, err := results.Aggregate()
+		k8s.EmitErrorEvent(r.recorder, err, &es, events.EventReconciliationError, "Reconciliation error: %v", err)
```

Comment:
This may lead to a big amount of events for errors we consider as "normal" (eg. apiserver optimistic locking conflicts).
Also will it lead to duplicating errors event? eg. I know an error is not supposed to happen so I emit an event deep down in the code, then this event gets triggered again in a different way here.